src/plugins/hapi.js

```diff
@@ -56,93 +57,62 @@ function createWrapDispatch (tracer, config) {
   }
 }
 
-module.exports = [
-  {
-    name: '@hapi/hapi',
-    versions: ['>=17.9'],
+function patchRequestGenerate (name, versions) {
```

Comment:
I know this boilerplate is inherited but the intermediate helpers don't provide much imo:

```
function patch (name, versions, file, property, method, prototypical) {
  return {
    name: name,
    versions: versions,
    file: file,
    patch (Obj, tracer, config) {
	    const target = prototypical ? Obj.prototype : Obj
      this.wrap(target, property, method(tracer, config))
    },
    unpatch (Obj) {
	    const target = prototypical ? Obj.prototype : Obj
      this.unwrap(target, property)
    }
  }
}

let patches = [
  patch('@hapi/hapi', ['>=17.9 <19.0'], 'lib/request.js', 'generate', createWrapGenerate, true),
  patch('hapi', ['>=17.1'], 'lib/request.js', 'generate', createWrapGenerate, true),
  patch('hapi', ['8.5 - 17.0'], 'lib/request.js', 'request', createWrapGenerate),
  patch('hapi', ['2 - 8.4'], 'lib/request.js', '_execute', createWrapExecute),
  patch('hapi', ['7.2 - 16'], 'lib/connection.js', '_dispatch', createWrapDispatch),
  patch('@hapi/hapi', ['>=17.9 <19.0'], 'lib/core.js', '_dispatch', createWrapDispatch),
  patch('hapi', ['>=17'], 'lib/core.js', '_dispatch', createWrapDispatch),
  patch('hapi', ['2 - 7.1'], 'lib/server.js', '_dispatch', createWrapDispatch)
]

if (semver.gte(process.version, '12.0.0')) {
  patches = patches.concat([
    patch('@hapi/hapi', ['>=19.0'], 'lib/request.js', 'generate', createWrapGenerate, true),
    patch('@hapi/hapi', ['>=19.0'], 'lib/core.js', '_dispatch', createWrapDispatch)
  ])
}
```

Reply:
I thought they made the array with `patch()` calls a bit more readable but I guess that goes into personal preference territory. Don't have any strong feelings about it. Can get rid of the intermediate ones.

Reply:
Updated.